example/models/food/v1/food_test.go

```diff
@@ -0,0 +1,138 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/ductone/protoc-gen-pgdb/internal/pgtest"
+	pgdb_v1 "github.com/ductone/protoc-gen-pgdb/pgdb/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSchemaFoodPasta(t *testing.T) {
+	ctx := context.Background()
+	pg, err := pgtest.Start()
+	require.NoError(t, err)
+	defer pg.Stop()
+
+	_, err = pg.DB.Exec(ctx, "CREATE EXTENSION IF NOT EXISTS btree_gin")
+	require.NoError(t, err)
+
+	smsg := &Pasta{}
+	schema, err := pgdb_v1.CreateSchema(smsg)
+	require.NoError(t, err)
+	for _, line := range schema {
+		fmt.Printf("%s \n", line)
+		_, err := pg.DB.Exec(ctx, line)
+		require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to execute sql: '\n%s\n'", line)
+		// os.Stderr.WriteString(line)
+		// os.Stderr.WriteString("\n------\n")
+	}
+	ct := schema[0]
+	require.Contains(t, ct, "CREATE TABLE")
+	require.Equal(t, 2,
+		strings.Count(ct, "$pksk"),
+		"Create table should contain only one pksk field + index: %s", ct,
+	)
+
+	require.Contains(t, ct, "PARTITION BY LIST")
+
+	require.Equal(t, 1,
+		strings.Count(ct, "fts_data"),
+		"Create table should contain only one fts_data field: %s", ct,
+	)
+	_, err = pg.DB.Exec(ctx, "DROP TABLE "+smsg.DBReflect().Descriptor().TableName())
+	require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to drop")
+
+	schema, err = pgdb_v1.Migrations(ctx, pg.DB, smsg)
+	require.NoError(t, err)
+	for _, line := range schema {
+		_, err := pg.DB.Exec(ctx, line)
+		require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to execute sql: '\n%s\n'", line)
+		// os.Stderr.WriteString(line)
+		// os.Stderr.WriteString("\n------\n")
+	}
+	ct = schema[0]
+	require.Contains(t, ct, "CREATE TABLE")
+
+	fakeTenantIds := []string{"t1", "t2", "t3"}
+	protoTableName := smsg.DBReflect().Descriptor().TableName()
+
+	// TODO(scott) makes helper functions and move to pgdb_v1
+	verifyMasterPartition(t, pg, protoTableName, fakeTenantIds)
+	// Test sub-tables for partitions
+	// Create sub-tables
+	for _, tenantId := range fakeTenantIds {
+		_, err = pg.DB.Exec(ctx, fmt.Sprintf("CREATE TABLE %s_%s PARTITION OF %s FOR VALUES IN ('%s')", protoTableName, tenantId, protoTableName, tenantId))
+		require.NoError(t, err, "TestSchemaFoodPasta: failed to create partitioned table")
+	}
+	verifySubTables(t, pg, protoTableName, fakeTenantIds)
```

Comment:
this test should also INSERT into master table
1) verify a SELECT on the master works
2) verify a select on the child tables a) works for the correct tenant, 2) tenant A data is NOT in tenant B table

Reply:
Ah yes thank you for reminding me!

Reply:
Added here https://github.com/ductone/protoc-gen-pgdb/pull/63/commits/408f6b2028e454391acfd07747c6f2cff0a49bd2